packages/account/src/utils/consolidate-coins.ts

```diff
@@ -0,0 +1,241 @@
+    all.push(...coins);
+    after = coins.pop()?.id;
+  }
+
+  return { coins: all };
+};
+
+/** @hidden */
+const sortCoins = ({ coins }: { coins: Coin[] }) => coins.sort((a, b) => b.amount.cmp(a.amount));
+
+/** @hidden */
+const createOuputCoin = (opts: {
+  transactionId: string;
+  outputs: Output[];
+  baseAssetId: string;
+}): Coin => {
+  const { transactionId, outputs, baseAssetId } = opts;
+
+  const outputChangeIndex = outputs.findIndex(
+    (output) => output.type === OutputType.Change && output.assetId === baseAssetId
+  );
+  if (outputChangeIndex === -1) {
+    throw new FuelError(ErrorCode.UNKNOWN, 'No change output found');
+  }
+
+  const outputCoin = outputs[outputChangeIndex] as OutputChange;
+
+  // Format the UTXO ID
+  const outputIndexPadded = Number(outputChangeIndex).toString().padStart(4, '0');
+
+  return {
+    id: `${transactionId}${outputIndexPadded}`,
+    assetId: outputCoin.assetId,
+    amount: outputCoin.amount,
+    owner: new Address(outputCoin.to),
+    blockCreated: bn(0),
+    txCreatedIdx: bn(0),
+  };
+};
+
+export const consolidateCoins = async ({
+  account,
+  assetId,
+}: {
+  account: Account;
+  assetId: string;
+}) => {
+  const chainInfo = await account.provider.getChain();
+  const chainId = chainInfo.consensusParameters.chainId.toNumber();
+  const gasPrice = await account.provider.estimateGasPrice(10);
+  const maxInputs = chainInfo.consensusParameters.txParameters.maxInputs.toNumber();
+  const baseAssetId = await account.provider.getBaseAssetId();
+  const isBaseAsset = assetId === baseAssetId;
+
+  const batchSize = maxInputs;
+  const numberOfFundingCoins = maxInputs;
+
+  let funding: Coin[] = [];
+  let dust: Coin[] = [];
+
+  // We get the largest coin/s for funding purposes
+  if (isBaseAsset) {
+    const coins = await getAllCoins(account, baseAssetId).then(sortCoins);
+    funding = coins.slice(0, numberOfFundingCoins);
+    dust = coins.slice(numberOfFundingCoins);
+  } else {
+    funding = await getAllCoins(account, baseAssetId)
+      .then(sortCoins)
+      .then((coins) => coins.slice(0, numberOfFundingCoins));
+    dust = await getAllCoins(account, assetId).then(({ coins }) => coins);
+  }
+
+  // There a better way of detecting whether the account has enough funds to consolidate
+  if (funding.length === 0) {
+    throw new FuelError(
+      ErrorCode.INSUFFICIENT_FUNDS,
+      `Insufficient funds to consolidate.\n\tAsset ID: ${baseAssetId}\n\tOwner: ${account.address.toB256()}`
+    );
+  }
+
+  const batches = [
+    ...splitEvery(batchSize, funding),
+    // We leave one coin for the funding coin
+    ...splitEvery(batchSize - 1, dust),
+  ];
+
+  const txs: ScriptTransactionRequest[] = batches.map((batch) => {
+    const request = new ScriptTransactionRequest({
+      scriptData: '0x',
+    });
+
+    // Add our dust coins as inputs
+    request.addResources(batch);
+
+    return request;
+  });
+
+  const submitAll: SubmitAllCallback = async (opts = {}) => {
```

Comment:
as the txs are being sent sequentially, why do we need the funding / dust logic? 

we could simplify by doing like this?
1- get all coins
2- sort bigger coins first
3- put 255 coins in the tx
4- put 1 output
5- wait tx success
6- repeat until user as only 1 coin